frontend/src/pages/transactions/[hash].tsx

```diff
@@ -98,18 +104,127 @@ const TransactionDetailsPage: NextPage<Props> = (props) => {
   );
 };
 
+export type TransactionOutcome = {
+  id: string;
+  outcome: RpcOutcome;
+  block_hash: string;
+};
+
+type ReceiptExecutionOutcome = {
+  tokens_burnt: string;
+  logs: string[];
+  outgoing_receipts?: NestedReceiptWithOutcome[];
+  status: RpcReceiptStatus;
+  gas_burnt: number;
+};
+
+export type NestedReceiptWithOutcome = {
+  actions?: Action<keyof RpcAction>[];
+  block_hash: string;
+  outcome: ReceiptExecutionOutcome;
+  predecessor_id: string;
+  receipt_id: string;
+  receiver_id: string;
+};
+
+export type Transaction = TransactionBaseInfo & {
+  status: ExecutionStatus;
+  receiptsOutcome: RpcReceiptOutcome[];
+  transactionOutcome: TransactionOutcome;
+  receipt: NestedReceiptWithOutcome;
+};
+
 export const getServerSideProps: GetServerSideProps<
   Props,
   { hash: string }
 > = async ({ req, params }) => {
   const hash = params!.hash;
   try {
+    const nearNetwork = getNearNetwork(req);
+    const transactionBaseInfo = await wampApi.call(
```

Comment:
This convertation is a big mess, we need to find out the way to make it more clear.